langchain/chat_models/azure_openai.py

```diff
@@ -121,12 +121,22 @@ def _identifying_params(self) -> Mapping[str, Any]:
         return {**self._default_params}
 
     @property
-    def _invocation_params(self) -> Mapping[str, Any]:
+    def _client_params(self) -> Dict[str, Any]:
+        """Get the config params used for the openai client."""
         openai_creds = {
             "api_type": self.openai_api_type,
             "api_version": self.openai_api_version,
         }
-        return {**openai_creds, **super()._invocation_params}
+        return {**super()._client_params, **openai_creds}
+
```

Comment:
Weired that we had `_invocation_params` and `_get_invocation_params` that each served different purposes:
1. `_get_invocation_params` was used in callbacks and for cache checking
2. `_invocation_params` used for actually passing params to the openai and azure openai client

Some other options would be:
- Always do `_get_invocation_params` but then either don't call `super()` or change the `self.dict()` call to filter out the `_type` arguments 
- Always do `_get_invocation_params` but then add the api key in a separate method